### ISSUE TITLE
Support for pydantic-like Models?

### ISSUE THREAD

**Issue by collerek** [koxudaxi/pydantic-pycharm-plugin#209]
Hi, I'm using your plugin for quite some time and it's awesome - thanks for your work! :) 

I'm developing ormar - a mini async orm with fastapi in mind, that uses pydantic for validation (https://github.com/collerek/ormar)

I'm subclassing the `pydanic.BaseModel` as my `ormar.Model` and hacking around with pydantic internals to make it work.
(You can ignore Meta class below, the field definition is like in pydantic but I use my custom fields classes as type hints - id and name in example).

![image](https://user-images.githubusercontent.com/16324238/97419174-0a69b800-190a-11eb-9aa9-518c60541ae7.png)
It does kind of work, but in example above it shows Base clases of my fields, and I would like to display `user=User(id: Integer, name: String)`, or even better -> `user=User(id:int, name:str)`

I'm already playing with `__annotations__`, pydantic fields `shape`, `type_` and a lot of internals, so I would like to adjust my models/ fields to play well with your plugin.

But for that I need information **from where do you grab the information to display in code completion, hints** etc.

I'm also exposing QuerySet operations like `User.objects.create(**kwargs)` or `User.objects.update(**kwargs)` that as kwargs should accept only pydantic/ormar fields -> can you somehow register a custom function to be picked up by your plugin, or make it work in other way (so display hints like in `__init__()`).

I tried looking into your code but saw a huge number of jetbeans dependencies and I don't even know kotlin :D 

So would be really grateful for info how I can adjust my project to be more in line with your plugin, as I find it super useful.

Thanks in advance and sorry for the long text.

**Comment by koxudaxi** [comment]
Thank you for posting a suggestion.
I feel your idea is great.

I have two ideas to resolve your problem.
1. you use Annotated (https://www.python.org/dev/peps/pep-0593/)
2. acceptable-type https://koxudaxi.github.io/pydantic-pycharm-plugin/type-checker-for-pydantic/#acceptable-type

1. is a standard feature in Python. you can use this way for other editor and mypy.
2. is an option for this plugin.
I don't use this way for your module. And I don't know to able to work it well. :thinking: 
I may need to change the code of this plugin.

**Comment by collerek** [comment]
Yeah, I realized that it does not play well with `mypy` and other static compilers anyway thanks to https://github.com/collerek/ormar/issues/25.

Saw that `pydantic` had the same problems and for complicated types they switched to notation like this:
```
class A(BaseModel):
    field1: int = Field(...)
```

So probably I will play around and see if I can subclass `FieldInfo` and also switch to something like:
```
class A(ormar.Model):
    field1: int = ormar.Integer(primary_key=True)
```

If I manage to do this will it play nice with your plugin?

Also is there a way to include those other methods in your plugin? I assume I cannot do anything in python (like set a decorator with name you provide or something) that will cause your plugin to pick it up and propose the same argument for those methods as in `__init__()`?

**Comment by collerek** [comment]
Ok, so I redesigned the models definition in v0.4.0 and if you provide type hints all plays well with your plugin:
![image](https://user-images.githubusercontent.com/16324238/97802781-26d95d80-1c46-11eb-98e2-eecc07696eec.png)

So now I have only this question if I can somehow add other methods that should have exactly the same hints and arguments as the `__init__()`?

**Comment by koxudaxi** [comment]
@collerek 

This plugin expects subclasses of BaseModel for injecting `__init__()`.
If you want to get the type-hint on other methods then, I have to change the plugin.
I should provide a new config for the feature. 

Also, We should try it to check possible or impossible before implementation.

**Comment by collerek** [comment]
How can I check it?

**Comment by koxudaxi** [comment]
you need to write code for prototyping
The best is you run the plugin with a debugger.

I'm thinking of an approach.
I implemented a feature to treat [create_model](https://pydantic-docs.helpmanual.io/usage/models/#dynamic-model-creation) https://github.com/koxudaxi/pydantic-pycharm-plugin/issues/169
If your method is similar to `create_model`, we can go to the next step.

Did you tell me about the method?
Does the method have a pydantic model? the plugin needs to lookup the model.
Also, What does the method return? the class of model? or instantiated model class?

**Comment by abreumatheus** [comment]
@collerek Any updates on how to get completion working with ormar?
